### ISSUE TITLE
CVE-2018-1002202 not detected

### ISSUE THREAD

**Issue by gallifabio** [dependency-check/DependencyCheck#1532]
Hi, me and my company want to move from Snyk to Dependency-Check. ATM we're doing some test and I can't understand why some CVE are not detected in Dependency-Check. I use the following tool:

Dependency-Check Command Line version
```sh
dependency-check.bat -v
Dependency-Check Core version 3.3.2
```

I set-up a simple project with the following Maven pom.xml:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>com.gallifabio</groupId>
  <artifactId>zip4tester</artifactId>
  <version>1.0-SNAPSHOT</version>
  <packaging>jar</packaging>
  <dependencies>
    <dependency>
      <groupId>net.lingala.zip4j</groupId>
      <artifactId>zip4j</artifactId>
      <version>1.3.1</version>
    </dependency>
  </dependencies>
</project>
```

I launch the command line tool with the following arguments:
```sh
dependency-check.bat --scan "D:\Sources\zip4tester\target" --project zip4tester
```

The "dependency-check-report.html" is being generated in the /target folder and the zip4j-1.3.1.jar is listed in the "Suppressed Vulnerabilities" section for another CVE:

[CVE-2018-13684](http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2018-13684)

Since the CVE-2018-1002202 is listed in the [NIST](https://nvd.nist.gov/vuln/detail/CVE-2018-1002202) I was expecting to see it in the "Vulnerable Dependencies" section since the zip4j version is < 1.3.3. Am I missing anything? Any help would be appreciated, TIA

**Comment by jeremylong** [comment]
This is occurring because the project needs to move from the NVD's XML data feeds to the JSON data feeds. This is high on our priority list (see #707).  This was delayed because the data feeds had been labeled "beta". I will be working on this as soon as I push 3.3.3 (hopefully tomorrow).

#### Details for the specific finding above are below.

The XML format does not contain the vulnerable software section:
```xml
  <entry id="CVE-2018-1002202">
    <vuln:vulnerable-configuration id="http://nvd.nist.gov/">
      <cpe-lang:logical-test operator="OR" negate="false"/>
    </vuln:vulnerable-configuration>
    <vuln:cve-id>CVE-2018-1002202</vuln:cve-id>
    <vuln:published-datetime>2018-07-25T13:29:00.797-04:00</vuln:published-datetime>
    <vuln:last-modified-datetime>2018-09-18T14:12:35.880-04:00</vuln:last-modified-datetime>
    <vuln:cvss>
      <cvss:base_metrics>
        <cvss:score>4.3</cvss:score>
        <cvss:access-vector>NETWORK</cvss:access-vector>
        <cvss:access-complexity>MEDIUM</cvss:access-complexity>
        <cvss:authentication>NONE</cvss:authentication>
        <cvss:confidentiality-impact>NONE</cvss:confidentiality-impact>
        <cvss:integrity-impact>PARTIAL</cvss:integrity-impact>
        <cvss:availability-impact>NONE</cvss:availability-impact>
        <cvss:source>http://nvd.nist.gov</cvss:source>
        <cvss:generated-on-datetime>2018-09-18T13:21:34.330-04:00</cvss:generated-on-datetime>
      </cvss:base_metrics>
    </vuln:cvss>
    <vuln:cwe id="CWE-22"/>
    <vuln:references xml:lang="en" reference_type="VENDOR_ADVISORY">
      <vuln:source>MISC</vuln:source>
      <vuln:reference href="https://github.com/snyk/zip-slip-vulnerability" xml:lang="en">https://github.com/snyk/zip-slip-vulnerability</vuln:reference>
    </vuln:references>
    <vuln:references xml:lang="en" reference_type="VENDOR_ADVISORY">
      <vuln:source>MISC</vuln:source>
      <vuln:reference href="https://snyk.io/research/zip-slip-vulnerability" xml:lang="en">https://snyk.io/research/zip-slip-vulnerability</vuln:reference>
    </vuln:references>
    <vuln:references xml:lang="en" reference_type="VENDOR_ADVISORY">
      <vuln:source>MISC</vuln:source>
      <vuln:reference href="https://snyk.io/vuln/SNYK-JAVA-NETLINGALAZIP4J-31679" xml:lang="en">https://snyk.io/vuln/SNYK-JAVA-NETLINGALAZIP4J-31679</vuln:reference>
    </vuln:references>
    <vuln:summary>zip4j before 1.3.3 is vulnerable to directory traversal, allowing attackers to write to arbitrary files via a ../ (dot dot slash) in a Zip archive entry that is mishandled during extraction. This vulnerability is also known as 'Zip-Slip'.</vuln:summary>
  </entry>
```

Whereas the JSON format does include the vulnerable software in the configuration section:
```json
{
    "cve" : {
      "data_type" : "CVE",
      "data_format" : "MITRE",
      "data_version" : "4.0",
      "CVE_data_meta" : {
        "ID" : "CVE-2018-1002202",
        "ASSIGNER" : "cve@mitre.org"
      },
      "affects" : {
        "vendor" : {
          "vendor_data" : [ ]
        }
      },
      "problemtype" : {
        "problemtype_data" : [ {
          "description" : [ {
            "lang" : "en",
            "value" : "CWE-22"
          } ]
        } ]
      },
      "references" : {
        "reference_data" : [ {
          "url" : "https://github.com/snyk/zip-slip-vulnerability",
          "name" : "https://github.com/snyk/zip-slip-vulnerability",
          "refsource" : "MISC",
          "tags" : [ "Exploit", "Third Party Advisory" ]
        }, {
          "url" : "https://snyk.io/research/zip-slip-vulnerability",
          "name" : "https://snyk.io/research/zip-slip-vulnerability",
          "refsource" : "MISC",
          "tags" : [ "Exploit", "Issue Tracking", "Patch", "Third Party Advisory" ]
        }, {
          "url" : "https://snyk.io/vuln/SNYK-JAVA-NETLINGALAZIP4J-31679",
          "name" : "https://snyk.io/vuln/SNYK-JAVA-NETLINGALAZIP4J-31679",
          "refsource" : "MISC",
          "tags" : [ "Third Party Advisory" ]
        } ]
      },
      "description" : {
        "description_data" : [ {
          "lang" : "en",
          "value" : "zip4j before 1.3.3 is vulnerable to directory traversal, allowing attackers to write to arbitrary files via a ../ (dot dot slash) in a Zip archive entry that is mishandled during extraction. This vulnerability is also known as 'Zip-Slip'."
        } ]
      }
    },
    "configurations" : {
      "CVE_data_version" : "4.0",
      "nodes" : [ {
        "operator" : "OR",
        "cpe" : [ {
          "vulnerable" : true,
          "cpe22Uri" : "cpe:/a:lingala:zip4j",
          "cpe23Uri" : "cpe:2.3:a:lingala:zip4j:*:*:*:*:*:*:*:*",
          "versionEndExcluding" : "1.3.3"
        } ]
      } ]
    },
    "impact" : {
      "baseMetricV3" : {
        "cvssV3" : {
          "version" : "3.0",
          "vectorString" : "CVSS:3.0/AV:L/AC:L/PR:N/UI:R/S:U/C:N/I:H/A:N",
          "attackVector" : "LOCAL",
          "attackComplexity" : "LOW",
          "privilegesRequired" : "NONE",
          "userInteraction" : "REQUIRED",
          "scope" : "UNCHANGED",
          "confidentialityImpact" : "NONE",
          "integrityImpact" : "HIGH",
          "availabilityImpact" : "NONE",
          "baseScore" : 5.5,
          "baseSeverity" : "MEDIUM"
        },
        "exploitabilityScore" : 1.8,
        "impactScore" : 3.6
      },
      "baseMetricV2" : {
        "cvssV2" : {
          "version" : "2.0",
          "vectorString" : "(AV:N/AC:M/Au:N/C:N/I:P/A:N)",
          "accessVector" : "NETWORK",
          "accessComplexity" : "MEDIUM",
          "authentication" : "NONE",
          "confidentialityImpact" : "NONE",
          "integrityImpact" : "PARTIAL",
          "availabilityImpact" : "NONE",
          "baseScore" : 4.3
        },
        "severity" : "MEDIUM",
        "exploitabilityScore" : 8.6,
        "impactScore" : 2.9,
        "obtainAllPrivilege" : false,
        "obtainUserPrivilege" : false,
        "obtainOtherPrivilege" : false,
        "userInteractionRequired" : true
      }
    },
    "publishedDate" : "2018-07-25T17:29Z",
    "lastModifiedDate" : "2018-09-18T18:12Z"
  }
```

**Comment by jeremylong** [comment]
5.0.0 has been released which should contain a fix for this issue.
